tools/apiview/parsers/csharp-api-parser/CSharpAPIParser/Program.cs

```diff
@@ -204,6 +205,122 @@ static bool IsDll(string name)
         return name.EndsWith(".dll", StringComparison.OrdinalIgnoreCase);
     }
 
+    /// <summary>
+    /// Parses the target framework from a DLL path within a NuGet package.
+    /// </summary>
+    /// <param name="path">The path to the DLL within the package (e.g., "lib/net8.0/System.Text.Json.dll")</param>
+    /// <returns>The target framework moniker or null if not found</returns>
+    public static string? ParseTargetFrameworkFromPath(string path)
+    {
+        // Expected format: lib/{tfm}/{assembly}.dll or ref/{tfm}/{assembly}.dll
+        var parts = path.Split('/', StringSplitOptions.RemoveEmptyEntries);
+        if (parts.Length >= 3 && (parts[0] == "lib" || parts[0] == "ref"))
+        {
+            return parts[1];
+        }
+        return null;
+    }
+
+    /// <summary>
+    /// Selects the best target framework from the available options.
+    /// Prefers newer and more specific frameworks.
+    /// </summary>
+    /// <param name="availableFrameworks">Available target framework monikers</param>
+    /// <returns>The best target framework moniker</returns>
+    public static string? SelectBestTargetFramework(IEnumerable<string> availableFrameworks)
```

Comment:
why is this algorithm so complex, using LINQ, sorting, etc. Just create an array of framework monikers in the order of preferences. then iterate over the array and see if the next preferred item exists in the passed in "avaliableFrameworks" list.